### ISSUE TITLE
Clarify "module" vs. "package"

### ISSUE THREAD

**Issue by mdboom** [tacaswell/version_pep_draft#6]
The text in #5 already does a good job of disambiguating between a distribution package and an imported package, and is precise about which of the overloaded terms it uses.

One place where I feel it's a bit unclear is its use of module vs. imported package, which are sometimes (but not always) the same thing.  I don't think I'm disagreeing with any of the *intent* in the language, just wanting to make it more precise somehow.  I don't know exactly how to resolve it, so I'm creating a discussion instead.

Saying "all modules should have a `__version__` doesn't seem quite correct.  While I'd expect `matplotlib.__version__` to exist, I wouldn't expect `matplotlib.artists.__version__` to have one (and that seems overly burdensome and unnecessary).  

So here, we really mean "all `__init__.py` modules in top-level packages should have a `__version__`.

But we also want single-file modules installed through some side channel and scripts to have a `__version__`.  These are more precisely "modules".

So while I think it's pretty intuitive what we want, I'm not sure how to word it: "all standalone modules and the top-level of every importable package" is awfully wordy.

**Comment by mdboom** [comment]
...oh and let's not forget namespace packages... (which the PEP already addresses, but should be part of this whole "defining what we mean by module" section)

**Comment by ChrisBarker-NOAA** [comment]
yeah, I've struggled with that a lot -- I wish the term pacakge didn't exist -- instead we have a module and and optionally sub-modules.

In practice, from the user's end, that's what we have: when you:

`import something`

You get a module -- whether it was imported from a dir with an `__init__.py`, or a single python file, or a compiled extension -- it's all the same thing.

>  "all standalone modules and the top-level of every importable package" is awfully wordy.

yeah, but maybe that's required -- at least in one place.

maybe an example, ,e.g.:

```
>>> import a_module
>>> a_module.__version__ # exists
'2.0.9'
```
vs:

```
>>> from a_module import sub_module
>>> sub_module.__version__
AttributeError: module 'submodule' has no attribute '__version__'
```

Side note: the standard library json module has a `__version__` string! 

```
>>> import json
>>> json.__version__
'2.0.9'
```
